File: android/src/main/java/app/capgo/plugin/textinteraction/TextInteractionPlugin.java
package app.capgo.plugin.textinteraction;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "TextInteraction")
public class TextInteractionPlugin extends Plugin {

    private final String pluginVersion = "8.0.11";

    private TextInteraction implementation = new TextInteraction();

    @PluginMethod
    public void toggle(PluginCall call) {
        boolean enabled = call.getBoolean("enabled", false);

        JSObject ret = new JSObject();
        ret.put("success", implementation.toggle(enabled));
        call.resolve(ret);
    }

    @PluginMethod
    public void getPluginVersion(final PluginCall call) {
        try {
            final JSObject ret = new JSObject();
            ret.put("version", this.pluginVersion);
            call.resolve(ret);
        } catch (final Exception e) {
            call.reject("Could not get plugin version", e);
        }
    }
}

File: ios/Sources/TextInteractionPlugin/TextInteractionPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(TextInteractionPlugin)
public class TextInteractionPlugin: CAPPlugin, CAPBridgedPlugin {
    private let pluginVersion: String = "8.0.5"
    public let identifier = "TextInteractionPlugin"
    public let jsName = "TextInteraction"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "toggle", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getPluginVersion", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = TextInteraction()

    @objc func toggle(_ call: CAPPluginCall) {
        let enabled = call.getBool("enabled") ?? false

        DispatchQueue.main.async {
            let success = self.implementation.toggle(enabled, webView: self.bridge?.webView)
            call.resolve([
                "success": success
            ])
        }
    }

    @objc func getPluginVersion(_ call: CAPPluginCall) {
        call.resolve(["version": self.pluginVersion])
    }

}
